tests/integration/index/create_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"testing"
+
+	"github.com/sourcenetwork/defradb/request/graphql/schema"
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestIndexCreateWithCollection_ShouldNotHinderQuerying(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Creation of index with collection should not hinder querying",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						Name: String @index
+						Age: Int
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 0,
+				// bae-52b9170d-b77a-5887-b877-cbdbb99b009f
+				Doc: `
+					{
+						"Name":	"John",
+						"Age":	21
+					}`,
+			},
+			testUtils.Request{
+				Request: `
+					query  {
+						Users {
+							_key
+							Name
+							Age
+						}
+					}`,
+				Results: []map[string]any{
+					{
+						"_key": "bae-52b9170d-b77a-5887-b877-cbdbb99b009f",
```

Comment:
suggestion: I don't see much value added here by including the key in the query, and it will mean an extra location to change if/when the dockey changes.  Suggest removal of the `_key` field.

Same for `TestIndexCreate_ShouldNotHinderQuerying` and `TestIndexDrop_ShouldNotHinderQuerying`